packages/e2e-tests/src/puppeteerEnvironment.js

```diff
@@ -39,15 +40,54 @@ class PuppeteerEnvironment extends OriginalEnvironment {
 
   async handleTestEvent(event, state) {
     if (event.name === 'test_fn_failure') {
-      const testName = state.currentlyRunningTest.name;
-      await this.storeArtifacts(testName);
+      const testName = `${state.currentlyRunningTest.parent.name}  ${state.currentlyRunningTest.name}`;
+      const errors = state.currentlyRunningTest.errors;
+      let errorMessages = '';
+      errors.forEach((error) => {
+        errorMessages += testName + '::' + error + '\n\n';
+      });
+
+      await this.storeArtifacts(testName, errorMessages);
+    }
+
+    // print error to console
+    const ignoredEvents = [
+      'setup',
+      'add_hook',
+      'start_describe_definition',
+      'add_test',
+      'finish_describe_definition',
+      'run_start',
+      'run_describe_start',
+      'test_start',
+      'hook_start',
+      'hook_success',
+      'test_fn_start',
+      'test_fn_success',
+      'test_done',
+      'test_skip',
+      'run_describe_finish',
+      'run_finish',
+      'teardown',
+    ];
+    if (!ignoredEvents.includes(event.name)) {
+      // eslint-disable-next-line no-console
+      console.log(
+        new Date().toString() +
+          ' Unhandled event(' +
+          event.name +
+          '): ' +
+          util.inspect(event)
+      );
```

Comment:
This output format was OK for debugging, but if we wanna keep this, we should probably change it. Don't need the date for example, and I don't think we need to check for other events.


```js
if (event.name === 'test_fn_failure') {
  console.log(...)
}
```